File: ankor-core/src/main/java/at/irian/ankor/messaging/Message.java
package at.irian.ankor.messaging;

/**
 * @author Manfred Geiler
 */
@Deprecated
public abstract class Message {
    //private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(Message.class);

    private String senderId;
    private String modelId;
    private String messageId;

    /**
     * for deserialization only
     */
    protected Message() {}

    protected Message(String senderId, String modelId, String messageId) {
        this.senderId = senderId;
        this.modelId = modelId;
        this.messageId = messageId;
    }

    public String getSenderId() {
        return senderId;
    }

    public String getModelId() {
        return modelId;
    }

    public String getMessageId() {
        return messageId;
    }

    @Override
    public String toString() {
        return "Message{" +
               "messageId='" + messageId + '\'' +
               "}";
    }


    public abstract boolean isAppropriateListener(MessageListener listener);

    public abstract void processBy(MessageListener listener);
}

File: #ANKOR.net/Ankor.Core/Messaging/Message.cs

namespace Ankor.Core.Messaging {
	
	/// <summary>
	/// Description of Message.
	/// </summary>
	public abstract class Message {
		
		public string SenderId {get;  set;}
		public string ModelId {get; private set;}
		public string MessageId {get; private set;}
		public string Property { get; private set; }
		
		protected Message(string senderId, string modelId, string messageId, string property) {
			this.SenderId = senderId;
			this.ModelId = modelId;
			this.MessageId = messageId;
			this.Property = property;
		}

		protected internal Message() { }

		public override string ToString() {
			return string.Format("SenderId: {0}, ModelId: {1}, MessageId: {2}, Property: {3}", SenderId, ModelId, MessageId, Property);
		}
	}
}
